packages/@aws-cdk/aws-codeartifact/lib/repository.ts

```diff
@@ -0,0 +1,384 @@
+     */
+  readonly externalConnections?: ExternalConnection[],
+
+  /**
+     * Principal to associate allow access to the repository
+     * @default AccountRootPrincipal
+     */
+  readonly principal?: iam.IPrincipal
+
+  /**
+   * Principal to associate allow access to the repository
+   * @default Read/Write
+   */
+  readonly policyDocument?: iam.PolicyDocument
+}
+
+/**
+ * Properties for a new CodeArtifact repository policy restricting package manipulation
+ * @experimental
+ */
+export interface PolicyRepositoryPackage {
+  /**
+     * Package format
+     */
+  readonly packageFormat: string,
+  /**
+     * Package namespace
+     */
+  readonly packageNamespace: string,
+  /**
+     * Package name
+     */
+  readonly packageName: string,
+}
+
+/**
+ * A new CodeArtifacft repository
+ * @experimental
+ */
+export class Repository extends Resource implements IRepository {
+  /**
+     * Import an existing Repository provided an ARN
+     *
+     * @param scope The parent creating construct
+     * @param id The construct's name
+     * @param repositoryArn repository ARN (i.e. arn:aws:codeartifact:us-east-2:444455556666:repository/my-domain/my-repo)
+     */
+  public static fromRepositoryArn(scope: Construct, id: string, repositoryArn: string): IRepository {
+    return Repository.fromRepositoryAttributes(scope, id, { repositoryArn: repositoryArn });
+  }
+
+  /**
+   * Import an existing repository
+   */
+  public static fromRepositoryAttributes(scope: Construct, id: string, attrs: RepositoryAttributes): IRepository {
+    const stack = Stack.of(scope);
+    const parsed = stack.parseArn(attrs.repositoryArn || '');
+
+    const spl = parsed.resourceName?.split('/') || [undefined, undefined];
+    const repositoryName = spl[1];
+    const domainName = spl[0];
+
+    if (!repositoryName || repositoryName == '') {
+      throw new Error('\'RepositoryName\' is required and cannot be empty');
+    }
+
+    if (!domainName || domainName == '') {
+      throw new Error('Domain name is required with the ARN');
+    }
+
+    class Import extends Resource implements IRepository {
+      repositoryDescription?: string | undefined;
+      repositoryDomainName?: string | undefined;
+      repositoryArn = attrs.repositoryArn;
+      repositoryName = repositoryName as string;
+      repositoryDomainOwner = parsed.account || '';
+
+      grantRead(_identity: iam.IGrantable): iam.Grant {
+        throw new Error('Method not implemented.');
+      }
+      grantWrite(_identity: iam.IGrantable): iam.Grant {
+        throw new Error('Method not implemented.');
+      }
+      grantReadWrite(_identity: iam.IGrantable): iam.Grant {
+        throw new Error('Method not implemented.');
+      }
+
+      onEvent(_id: string, _options?: events.OnEventOptions | undefined): events.Rule {
+        throw new Error('Method not implemented.');
+      }
+      onPackageVersionStateChange(_id: string, _options?: events.OnEventOptions | undefined): events.Rule {
```

Comment:
Are these just TBD, or some underlying reason why they can't be implemented?

Reply:
Implemented